### ISSUE TITLE
Incompatibility with the carpet mod due to mixin priority collision

### ISSUE THREAD

**Issue by lesoup-mxd** [NivOridocs/flowstone#10]
both Carpet and Flowstone use the same mixin prio "1000" for LiquidBlock mixin

as a solution i suggest upping the prio of the mixin
@Mixin(value = LiquidBlock.class, priority = 1001)

yea i was too lazy for a pull request

**Comment by lesoup-mxd** [comment]
Maybe not that easy. getting:
Caused by: org.spongepowered.asm.mixin.injection.throwables.InvalidInjectionException: @At("INVOKE") on net/minecraft/class_2404::receiveFluidToDeepslate with priority 1000 cannot inject into net/minecraft/class_2404::method_10316(Lnet/minecraft/class_1937;Lnet/minecraft/class_2338;Lnet/minecraft/class_2680;)Z merged by niv.flowstone.mixin.LiquidBlockMixin with priority 1001 [INJECT_PREPARE Applicator Phase -> carpet.mixins.json:LiquidBlock_renewableDeepslateMixin from mod carpet -> Prepare Injections -> handler$zjc000$carpet$receiveFluidToDeepslate(Lnet/minecraft/class_1937;Lnet/minecraft/class_2338;Lnet/minecraft/class_2680;Lorg/spongepowered/asm/mixin/injection/callback/CallbackInfoReturnable;)V -> Prepare ->  -> { target: method_10316(Lnet/minecraft/class_1937;Lnet/minecraft/class_2338;Lnet/minecraft/class_2680;)Z }]
        at org.spongepowered.asm.mixin.injection.code.Injector.findTargetNodes(Injector.java:312) ~[sponge-mixin-0.15.2+mixin.0.8.7.jar:0.15.2+mixin.0.8.7]
        at org.spongepowered.asm.mixin.injection.code.Injector.find(Injector.java:248) ~[sponge-mixin-0.15.2+mixin.0.8.7.jar:0.15.2+mixin.0.8.7]
        at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.prepare(InjectionInfo.java:475) ~[sponge-mixin-0.15.2+mixin.0.8.7.jar:0.15.2+mixin.0.8.7]
        at org.spongepowered.asm.mixin.transformer.MixinTargetContext.prepareInjections(MixinTargetContext.java:1407) ~[sponge-mixin-0.15.2+mixin.0.8.7.jar:0.15.2+mixin.0.8.7]
        at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.prepareInjections(MixinApplicatorStandard.java:731) ~[sponge-mixin-0.15.2+mixin.0.8.7.jar:0.15.2+mixin.0.8.7]
        at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyMixin(MixinApplicatorStandard.java:315) ~[sponge-mixin-0.15.2+mixin.0.8.7.jar:0.15.2+mixin.0.8.7]
        at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:246) ~[sponge-mixin-0.15.2+mixin.0.8.7.jar:0.15.2+mixin.0.8.7]
        at org.spongepowered.asm.mixin.transformer.TargetClassContext.apply(TargetClassContext.java:437) ~[sponge-mixin-0.15.2+mixin.0.8.7.jar:0.15.2+mixin.0.8.7]
        at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:418) ~[sponge-mixin-0.15.2+mixin.0.8.7.jar:0.15.2+mixin.0.8.7]
        at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363) ~[sponge-mixin-0.15.2+mixin.0.8.7.jar:0.15.2+mixin.0.8.7]
        at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234) ~[sponge-mixin-0.15.2+mixin.0.8.7.jar:0.15.2+mixin.0.8.7]
        at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202) ~[sponge-mixin-0.15.2+mixin.0.8.7.jar:0.15.2+mixin.0.8.7]
        at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:422) ~[fabric-loader-0.16.3.jar:?]
        ... 12 more

**Comment by lesoup-mxd** [comment]
Carpet: LiquidBlock_renewableDeepslateMixin
Flowstone: niv.flowstone.mixin.LiquidBlockMixin

**Comment by lesoup-mxd** [comment]
Ah, seems like it launches at least, when setting prio to 999

**Comment by NivOridocs** [comment]
As a quick fix, I set the latest released Flowstone version to be incompatible with Carpet.

Meanwhile, I found a different solution than messing with priorities, and that is injecting Flowstone's logic at the HEAD of the conflicting method (which is LiquidBlock.shouldSpreadLiquid) instead of wholeheartedly overwriting it.

Notice that this solution makes it so that a couple of Carpet's functionalities (namely, renewableBlackstone and renewableDeepslate) may not work as intended or not work at all, which shouldn't be a problem since Flowstone can replace both, but let me know if it is.

Otherwise, I'll release a new version with the fix in a few days.
